src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -834,7 +834,8 @@ class VM_Version : public Abstract_VM_Version {
     return true;
   }
 
-  // For AVX CPUs only. f16c support is disabled if UseAVX == 0.
+  // For AVX CPUs only since it needs VEX encoding which is missing on SSE targets,
+  // thus f16c support is disabled if UseAVX == 0.
```

Comment:
This comment is somewhat or vey confusing. The code for supports_float16() by itself is very clear. I am wondering why do we need this explanation in the comment at all? Let us remove it altogether.

Reply:
Sure, FTR my comment was in relation to following check
https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/vm_version_x86.cpp#L1056

 FP16-FP32 conversions are VEX encoded instructions and do not have SSE flavor. Thus, only works with UseAVX > 0.